src/components/RecordMoviesComponent/index.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from "react";
+import classNames from "classnames";
+
+import RecordMoviesModal from "../RecordMoviesModal";
+import ViewportButton from "../ViewportButton";
+
+import styles from "./style.css";
+
+interface RecordMovieComponentProps {
+    movieUrl: string;
+    movieTitle: string;
+    cleanupMovieState: () => void;
+    startRecording: () => void;
+    stopRecording: () => void;
+}
+
+const RecordMovieComponent = (props: RecordMovieComponentProps) => {
+    const {
+        movieUrl,
+        movieTitle,
+        cleanupMovieState,
+        startRecording,
+        stopRecording,
+    } = props;
+    const [isRecording, setIsRecording] = useState<boolean>(false);
+    const [modalVisible, setModalVisible] = useState(false);
+    const [isHovering, setIsHovering] = useState(false);
+
+    const supportedBrowser = "VideoEncoder" in window;
+
+    const closeModal = () => {
+        cleanupMovieState();
+        setModalVisible(false);
+    };
+
+    const start = () => {
+        setIsRecording(true);
+        startRecording();
+    };
+
+    const stop = () => {
+        setIsRecording(false);
+        stopRecording();
+        setModalVisible(true);
+    };
+
+    const downloadMovie = () => {
+        if (!movieUrl) {
+            console.error("No recorded movie to download");
+            return;
+        }
+        const anchor = document.createElement("a");
+        anchor.href = movieUrl;
+        anchor.download = movieTitle;
+        document.body.appendChild(anchor);
+        anchor.click();
+        document.body.removeChild(anchor);
+        cleanupMovieState();
+    };
+
+    const startRecordIcon = (
+        <span className={styles.iconContainer}>
+            <span
+                className={classNames("icon-moon", "record-icon-circle")}
+            ></span>
+            <span
+                className={classNames("icon-moon", "record-icon-ring")}
+            ></span>
+        </span>
+    );
```

Comment:
To do multicolor icons via icomoon requires stacking glyphs. This was the best I could come up with, but I'm open to feedback if there is a better way!

Debated defining this in `Icons.tsx` with inline styles and importing from there, but that file typically doesn't handle any thing related to icomoon icons... Style rules for this icon also got a bit spread out, some in primary stylesheet, some here, and some in viewport button. Not ideal? Could consolidate somewhat by most of it to ViewportButton stylesheet but it seems a bit case-specific for that, so I only put the things that relied on disabled state there. This works, but I'm definitely down to refactor/move things around.